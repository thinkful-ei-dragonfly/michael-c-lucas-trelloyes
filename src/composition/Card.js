import React from 'react';
import './Card.css';

function Card(props) {
  return (
    <div className='Card'>
      <button type='button'>delete</button>
      <h3>{props.title}</h3>
      <p>{props.content}</p>
    </div>
  )
}

class CardClass extends React.Component {
  static defaultProps = {
    title: '',
    content: ''
  };
  render(){
    return (
      <div className='Card'>
        <button type='button'>delete</button>
        <h3>{this.props.title}</h3>
        <p>{this.props.content}</p>
      </div>
    )
  }
}

export default Card;
