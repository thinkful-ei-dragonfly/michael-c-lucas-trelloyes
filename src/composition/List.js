import React from 'react';
import './List.css';
import { spawn } from 'child_process';

function Tooltip(props) {
  return (
    <span className='Tooltip'>

    </span>
  )
}

export default Tooltip;
