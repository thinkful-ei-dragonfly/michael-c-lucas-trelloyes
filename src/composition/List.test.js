import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'

import List from './List'
import Card from './Card';

describe('<List />', () => {
  it('Renders List without failing [Snapshot Test]', () => {
    // Renders list as JSON
    // const cardArray = [<Card key={1} title='Title 1' content='Content 1'/>,<Card key={2} title='Title 2' content='Content 2'/> ];
    const cardArray = [{id: 1}, { id: 2}];
    const tree = renderer.create(<List cards={cardArray}/>).toJSON();
    // Checks whether it matches the previous Snapshot
    expect(tree).toMatchSnapshot();
  })
})
