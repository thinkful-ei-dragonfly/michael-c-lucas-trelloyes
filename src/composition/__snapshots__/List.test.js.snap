// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Renders List without failing [Snapshot Test] 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        
      </h3>
      <p>
        
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        
      </h3>
      <p>
        
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
